### PR TITLE
Cancels upload if the bundle uuid already exists in the storage system

### PR DESCRIPTION
**Pull request by GPelayo**
Cancels upload if the bundle/workflow/gnos uuid already exists in the storage system.

### CODE REVIEW

File: spinnaker.py [hunk 1]
```diff
@@ -899,6 +901,27 @@ def main():
...
+
+    # Checks if bundle_id/gnos_id exists in the
+    for fmo in flatMetadataObjs:
+        metadata_url = "https://metadata.{}/entities?gnosId={}".format(redwood_host, fmo['workflow_uuid'])
+        file_name_metadata_json = urlopen(metadata_url, context=ctx).read()
+        file_name_metadata = json.loads(file_name_metadata_json)
+        # pprint.pprint(file_name_metadata)
+        if file_name_metadata['totalElements'] > 0:
+            logging.error("Found row in manifest with the same sample_uuid, workflow_name, and workflow_version in the database.\n"
+                          "Please update the workflow version of that row.")
```
Comment (caaespin) [inline spinnaker.py]: Add instructions to bump up the workflow version number, 

What caused the problem
Why it caused the problem, 
How to fix it "The data in this row seems to have been uploaded previously. To uplaod again, please increase the workflow version for that row and re-upload. NO DATA WAS UPLOADED." something like that

File: spinnaker.py [hunk 2]
```diff
@@ -899,6 +901,27 @@ def main():
...
+            logging.error("Reference file metadata:\n{}\n\n".format(pprint.pformat(file_name_metadata)))
+            has_existing_bundle_id = True
+
+    if has_existing_bundle_id:
+        sys.exit(1)
```
Comment (caaespin) [inline spinnaker.py]: Maybe have a variable and accumulate the file names on the for loop before. Then print it right before the sys.exit call

File: spinnaker.py [hunk 3]
```diff
@@ -899,6 +901,27 @@ def main():
...
             flatMetadataObjs.append(metaObj)
 
+    redwood_host = os.environ['REDWOOD_ENDPOINT']
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+    has_existing_bundle_id = False
+
+    # Checks if bundle_id/gnos_id exists in the
+    for fmo in flatMetadataObjs:
```
Comment (caaespin) [inline spinnaker.py]: Try to add more comments for each significant block of code. e.g. what the for loop does, what each chunk within the for loop block does, etc.

**Review by caaespin — COMMENTED** [review]
Make sure the checksums are done after the changes you made. That way they don't have to wait a long time before they know if fails.

**Review by wshands — APPROVED** [review]
Looks OK to me.

**Review by caaespin — APPROVED** [review]
LGMT

image

**Pull request merged**